docs/articles/documentation/test-api/selecting-page-elements/selectors/selector-options.md

```diff
@@ -42,6 +42,59 @@ or the [selector-timeout](../../../using-testcafe/command-line-interface.md#--se
 
 `true` to additionally require the returned element to become visible within [options.timeout](#optionstimeout).
 
+This option is in effect when TestCafe waits for the selector to return a page element. This includes situations when
+
+* a property is obtained from the selector;
+
+    ```js
+    const width = await Selector('#element').with({ visibilityCheck: true }).clientWidth;
+    ```
+
+* a selector property is passed to an [assertion](../../assertions/README.md) as its actual value;
+
+    ```js
+    await t.expect(Selector('#element').with({ visibilityCheck: true }).clientWidth).eql(400);
+    ```
+
+* a selector is evaluated using the `await` keyword;
+
+    ```js
+    const snapshot = await Selector('#element').with({ visibilityCheck: true })();
+    ```
+
+If the target element is not visible, the selector throws an exception in all these cases.
+
+Note that when a selector is passed to a [test action](../../actions/README.md) as an identifier for the target element,
```

Comment:
Should we mention here [actions](https://github.com/DevExpress/testcafe/blob/3638f55ec4285630a6e8d5a2ae129727ec5eb429/src/test-run/commands/actions.js#L355) which don't check element visibility?

Reply:
What commands do you mean? Only `SetFilesToUploadCommand `?

Reply:
SetFilesToUploadCommand and ClearUploadCommand

Reply:
I'd better add this to https://devexpress.github.io/testcafe/documentation/test-api/waiting-for-page-elements-to-appear.html